### PR TITLE
Add shaderc_get_spv_version() to retrieve SPIR-V version/revision

### PR DESCRIPTION
**Pull request by mgadams**
This is useful for diagnostic purposes when embedded in test apps.

### CODE REVIEW

File: libshaderc/src/shaderc_test.cc [hunk 1]
```diff
@@ -52,6 +52,14 @@ TEST(Init, MultipleThreadsCalling) {
   shaderc_compiler_release(compiler3);
 }
 
+TEST(Init, SPVVersion) {
+  unsigned int version = 0;
+  unsigned int revision = 0;
+  shaderc_get_spv_version(&version, &revision);
+  EXPECT_TRUE(version == spv::Version);
```
Comment (dmircevski) [inline libshaderc/src/shaderc_test.cc]: Use EXPECT_EQ or EXPECT_THAT(version, Eq(spv::Version)).

Same below.

File: libshaderc/src/shaderc.cc [hunk 2]
```diff
@@ -23,6 +23,7 @@
 
 #include "libshaderc_util/compiler.h"
 #include "libshaderc_util/resources.h"
+#include "SPIRV/spirv.hpp"
```
Comment (dmircevski) [inline libshaderc/src/shaderc.cc]: Our style requires this be moved up, as "other libraries' .h file":

http://google-styleguide.googlecode.com/svn/trunk/cppguide.html#Names_and_Order_of_Includes

**Comment by mgadams** [comment]
Both updates done as requested.

File: libshaderc/src/shaderc_test.cc [hunk 3]
```diff
@@ -52,6 +52,14 @@ TEST(Init, MultipleThreadsCalling) {
   shaderc_compiler_release(compiler3);
 }
 
+TEST(Init, SPVVersion) {
+  unsigned int version = 0;
+  unsigned int revision = 0;
+  shaderc_get_spv_version(&version, &revision);
+  EXPECT_EQ(version, spv::Version);
```
Comment (dmircevski) [inline libshaderc/src/shaderc_test.cc]: Looks like you got the arguments swapped -- spv::Version is the expected, version the actual.

Comment (mgadams) [inline libshaderc/src/shaderc_test.cc]: Ah, I didn't realize order mattered (never used this test framework before). I'll send another update later this morning.

Comment (dmircevski) [inline libshaderc/src/shaderc_test.cc]: Yeah, it matters for correct failure messages -- it's pretty nifty when a failing test says something like
```
Value of: version
  Actual: 456
Expected: spv::Version
Which is: 123
```

Comment (mgadams) [inline libshaderc/src/shaderc_test.cc]: Definitely a useful feature. I've updated the commit accordingly.

**Pull request merged**